pkg/rancher-desktop/window/index.ts

```diff
@@ -167,6 +177,15 @@ export function openMain() {
     );
   }
 
+  window.on('closed', () => {
+    const cfg = loadSettings();
+
+    if (cfg.window.quitOnClose) {
+      closeAll();
```

Comment:
See above — the `closeAll()` is basically redundant (we don't need to close windows before quitting).

Reply:
We don't need to, but closing existing open windows provides more immediate feedback. Otherwise, existing windows like preferences hang around while we wait for Rancher Desktop to shut down.

Reply:
Hmm, looking at the code…
https://github.com/electron/electron/blob/d02c9f8bc644ba39acf9caa908c33a737202a790/shell/browser/browser.cc#L82-L94
It triggers `before-quit`, then closes all the windows.  We do a bunch of stuff there:
https://github.com/rancher-sandbox/rancher-desktop/blob/bf3825e1619edaaf3e79443887864d9bf3dd6fd5/background.ts#L380-L406
Most notably, we stop the backend (with the progress bar) first.  Except that this is the case where the main window was already closed, so there would be no visible feedback first.

Okay, I guess it makes sense to close all (other, non-main) windows first.